packages/net/writebuffer.pony

```diff
@@ -0,0 +1,69 @@
+use "collections"
+
+class WriteBuffer
+  """
+  A simple constant sized byte buffer to push Numbers into.
+  The data container is iso in order to keep it sendable(?)
+  Potential upgrades:
+  1. Have a list of data buffers of preset size
+  2. Arguably, seperate functions for _be and _le
+  3. A better way to get data out of the buffer, without passing a new array?
+  """
+  var data: Array[U8] iso
+  var _offset: USize
+  var _little_endian: Bool
+
+  new create(size: USize, little_endian: Bool = false) =>
+    """
+    Allocate the buffer
+    """
+    data = recover Array[U8].undefined(size) end
+    _offset = 0
+    _little_endian = little_endian
+
+  fun ref get_data(x: Array[U8] iso): Array[U8] iso^ =>
```

Comment:
@jemc you are right. i dont understand how this method really get used. With documentation perhaps I could understand. Its use in the tests in no way makes its usage more clear as it doesn't do anything with the data returned. from the tests it appears to be a setter not something that is designed for "get_data"